design/helm.md

```diff
@@ -0,0 +1,145 @@
+By allowing Kurl users to specifify a chart and its values, we can allow customization of kutl with new addons while delegating addons support and maintance to its community.
+
+## High-Level Design
+
+Kurl will bundle the Helm binary into the common tools bundle already delivering krew and kubectl plugins. The install.sh script will perform the following in the online case:
+1. Unarchive Helm and move into the `$PATH` and `/usr/bin`
+1. For each release, check if the release exists. 
+1. If the release exists, perform a `helm upgrade`.
+1. If the release doesn't exist, perform a `helm install`
+
+For airgapped the chart and artifacts need to be packaged into a bundle. The builder should perform the following:
+1. `helm pull` the repo
+1. `helm template` to render the chart with the provided values
+1. Generate an adhoc `manifest` file by parsing `image` specs from the rendered yaml.
+1. Generate a bundle including the chart + containers
+
+In the airgapped case, install.sh will perform the following:
+1. Unpack the helm bundle on initialization
+1. Load docker containers as part of the initialization
+1. Unarchive the chart and install with the installer spec values during the install phase
+
+## Detailed Design
+
+The yaml design is informed by the [Flux Helm Operator CRD](https://github.com/fluxcd/helm-operator/blob/master/docs/references/helmrelease-custom-resource.md). 
+Another spec is [Rancher's Helm CRD](/var/lib/rancher/rke2/server/manifests/) that works with k3s and RKE2.
+
+The proposal is to implement option1 in the updated Kurl Installer Spec:
+```yaml
+apiVersion: "cluster.kurl.sh/v1beta1"
+kind: "Installer"
+metadata: 
+  name: "latest"
+spec: 
+  kubernetes: 
+    version: "latest"
+  weave: 
+    version: "latest"
+  containerd: 
+    version: "latest"
+  helm:                 # Optional Future Capability
+    version: "latest"
+  helmReleases:
+    # Basic Idea, Option 1
+    - release: nginx-ingress
+      namespace: kube-system    # Optional
+      chart: 
+        name: bitnami/nginx
+        repo: https://charts.bitnami.com/bitnami 
+        version: 8.5.1          # Optional - uses latest if not provided
+      values: |                 # Optional
+        ingress:
+          enabled: true
+          hostname: dan-dot-com
+        metrics: 
+          enabled: true
+    # Option 2 (Using Mapstructure?)
+    - release: postgres-web
+      chart: 
+        name: bitnami/postgresql
+        repo: https://charts.bitnami.com/bitnami 
+      values:
+        postgresqlUsername: dan
+        postgresqlPassword: opensesame
+    # Option 3 (Config Url)
+    - release: postgres-auth
+      chart: 
+        name: bitnami/postgresql
+        repo: https://charts.bitnami.com/bitnami 
+      values: https://gist.githubusercontent.com/DanStough/b4376b7a6aa5b73c6a647bff504212df/raw/829680f4a474720dc36b7a963d2d691305b8cdfb/postgres-auth.yaml
```

Comment:
id opt for allow url or inline. in that case maybe two separate properties? `values` and `valuesUrl`?

Reply:
Agree.